framework/src/main/java/org/checkerframework/framework/util/typeinference8/constraint/ConstraintSet.java

```diff
@@ -190,12 +224,17 @@ public ConstraintSet getClosedSubset(Dependencies dependencies) {
       return subset;
     }
 
-    outDependencies.clear();
-    inputDependencies.clear();
+    // TODO: double check that this code is correct.
+    // checker/tests/all-systems/java8inference/MapEntryGetFails.java is a test that uses this code.
+    Set<Variable> inputDependencies = new LinkedHashSet<>();
+    Set<Variable> outDependencies = new LinkedHashSet<>();
     // If this subset is empty, then there is a cycle (or cycles) in the graph of dependencies
     // between constraints.
+    // In this case, the constraints in C that participate in a dependency cycle (or cycles) and do
+    // not depend on any constraints outside of the cycle (or cycles) are considered.
```

Comment:
What does "are considered" mean?  Please avoid passive voice and also use an active verb to clarify exactly what is done.